velox/functions/prestosql/types/IPPrefixType.cpp

```diff
@@ -120,6 +128,30 @@ class IPPrefixCastOperator : public exec::CastOperator {
     });
   }
 
+  static void castFromIpAddress(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* rowVectorResult = result.as<RowVector>();
+    const auto* ipAddrVector = input.as<SimpleVector<int128_t>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      auto intIpAddr = ipAddrVector->valueAt(row);
+      const auto tryPrefixLength =
+          ipaddress::tryIpPrefixLengthFromIPAddressType(intIpAddr);
+      if (FOLLY_UNLIKELY(tryPrefixLength.hasError())) {
+        context.setStatus(row, std::move(tryPrefixLength.error()));
+        return;
+      }
+
+      rowVectorResult->childAt(0)->as<FlatVector<int128_t>>()->set(
```

Comment:
Move type casts out of the per-row loop.  You can also use `asChecked` once it's outside loop.